### PR TITLE
chore: added test case for envName and envId

### PR DESCRIPTION
**Pull request by sondermanish**
Fixes https://github.com/appsmithorg/appsmith/issues/25872

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/DatasourceAnalyticsUtils.java [hunk 1]
```diff
@@ -88,4 +97,19 @@ public static Map<String, Object> getAnalyticsPropertiesWithStorage(
...
+            DatasourceStorage datasourceStorage, String dsCreatedAt, String environmentName) {
+        Map<String, Object> data = new HashMap<>();
+
+        data.put(DATASOURCE_ID_SHORTNAME, ObjectUtils.defaultIfNull(datasourceStorage.getDatasourceId(), ""));
+        data.put(ENVIRONMENT_ID_SHORTNAME, ObjectUtils.defaultIfNull(datasourceStorage.getEnvironmentId(), ""));
+        data.put(DATASOURCE_NAME_SHORTNAME, datasourceStorage.getName());
+        data.put(DATASOURCE_IS_TEMPLATE_SHORTNAME, ObjectUtils.defaultIfNull(datasourceStorage.getIsTemplate(), ""));
+        data.put(DATASOURCE_IS_MOCK_SHORTNAME, ObjectUtils.defaultIfNull(datasourceStorage.getIsMock(), ""));
+        data.put(DATASOURCE_CREATED_AT_SHORTNAME, dsCreatedAt);
+        data.put(ENVIRONMENT_NAME_SHORTNAME, ObjectUtils.defaultIfNull(environmentName, ENVIRONMENT_NAME_DEFAULT));
```
[the remark this answers is not in the document]
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/DatasourceAnalyticsUtils.java:111]: it's a default string constant ` ""`

[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/DatasourceAnalyticsUtils.java:111]: Why do we need the specific constant though?

[the remark this answers is not in the document]
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/DatasourceAnalyticsUtils.java:111]: Not specifically required, just kept it to be used as default when nothing is found

[the remark this answers is not in the document]
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/DatasourceAnalyticsUtils.java:111]: should i change it to a locally declared string?

**Review by tkAppsmith — APPROVED** [review]

**Pull request merged**